codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules2/AssignTypesVisitor.java

```diff
@@ -90,12 +88,6 @@ public RuleExpression visitFunctionCallExpression(FunctionCallExpression e) {
             return expr;
         }
 
-        List<RuleType> args = expr.arguments().stream().map(RuleExpression::type).collect(Collectors.toList());
-        if (!function.matches(args)) {
-            addError("Arguments for function `%s` doesn't match, expected: `%s`, got: `%s`",
-                     name,
-                     function.arguments().values(), args);
-        }
```

Comment:
We need to remove this error check here and above since they will fail when there is two assigns (LetExpression) in a row where second assign depends on the first assigned variable.

Because of the order of execution of the visitors, the first variable will not be stored in SymbolTable.locals before the second assign executes initially. The second assign will execute again successfully after the first variable is stored when super.visitXX is invoked